client/state/signup/steps/survey/actions.js

```diff
@@ -2,10 +2,26 @@
  * Internal dependencies
  */
 import { SIGNUP_STEPS_SURVEY_SET } from 'state/action-types';
+import {
+	composeAnalytics,
+	recordCustomAdWordsRemarketingEvent,
+	recordCustomFacebookConversionEvent,
+	joinAnalytics
+} from 'state/analytics/actions';
 
 export function setSurvey( survey ) {
-	return {
-		type: SIGNUP_STEPS_SURVEY_SET,
-		survey
-	};
+	const vertical = survey.vertical;
+	const attributes = { vertical };
+	const compositeAnalytics = composeAnalytics(
+		recordCustomAdWordsRemarketingEvent( attributes ),
+		recordCustomFacebookConversionEvent( 'VerticalSelectedEvent', attributes )
+	);
```

Comment:
I'm not sure if any existing code is using `composeAnalytics()` directly but it wasn't originally intended to be used - rather `withAnalytics()` was the intention. may not be troublesome, but no guarantees were made with the calling convention for `composeAnalytics()`